web-common/src/features/dashboards/time-dimension-details/charts/patch-vega-spec.ts

```diff
@@ -0,0 +1,151 @@
+import { COMPARIONS_COLORS } from "@rilldata/web-common/features/dashboards/config";
+import {
+  MainAreaColorGradientDark,
+  MainLineColor,
+} from "@rilldata/web-common/features/dashboards/time-series/chart-colors";
+import { TIME_GRAIN } from "@rilldata/web-common/lib/time/config";
+import { V1TimeGrain } from "@rilldata/web-common/runtime-client";
+import { VisualizationSpec } from "svelte-vega";
+import { TDDAlternateCharts, TDDChart } from "../types";
+
+function patchSpecForTimeComparison(
+  sanitizedSpec,
+  chartType: TDDAlternateCharts,
+  timeUnit: string,
+  measureName: string,
+  yEncoding,
+  colorEncoding,
+) {
+  yEncoding.field = "measure";
+
+  sanitizedSpec.transform = [
+    // Sanitize and transform comparison data in the right time format
+    { timeUnit: timeUnit, field: "comparison\\.ts", as: "comparison_ts" },
+    // Expand datum to have a key field to differentiate between current and comparison data
+    { fold: ["ts", "comparison_ts"] },
+    // Add a measure field to hold the right measure value
+    {
+      calculate: `(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])`,
```

Comment:
**Security Issue**: The dynamic data insertion into the 'calculate' field without proper sanitization or encoding can lead to injection vulnerabilities, such as XSS or code injection. <br> **Fix**: Ensure that any dynamic values inserted into the Vega-Lite specification are properly sanitized to prevent injection attacks. <br> **Code Suggestion**: 
 ```
 -      calculate: '(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])',
 +      calculate: 'sanitize(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])',
 ```